cmd/kapacitord/main.go

```diff
@@ -88,16 +89,44 @@ func (m *Main) Run(args ...string) error {
 		}
 
 		signalCh := make(chan os.Signal, 1)
-		signal.Notify(signalCh, os.Interrupt, syscall.SIGTERM)
+		signal.Notify(signalCh, os.Interrupt, syscall.SIGTERM, syscall.SIGHUP)
 		m.Logger.Println("I! Listening for signals")
 
 		// Block until one of the signals above is received
-		select {
-		case <-signalCh:
-			m.Logger.Println("I! Signal received, initializing clean shutdown...")
-			go func() {
-				cmd.Close()
-			}()
+	Loop:
+		for {
+			select {
+
+			case s := <-signalCh:
+				switch s.String() {
+				case syscall.SIGTERM.String():
+					m.Logger.Println("I! SIGTERM received, initializing clean shutdown...")
+					go func() {
+						cmd.Close()
+					}()
+					break Loop
+
+				case syscall.SIGHUP.String():
+					m.Logger.Println("I! SIGHUP received, Reloading tasks/templates/handlers directory...")
+					if err := cmd.Server.LoadService.Load(); err != nil {
+						m.Logger.Println(fmt.Sprintf("E! Failed to reload tasks/templates/handlers: %s", err))
+						if _, ok := err.(load.HardError); ok {
```

Comment:
What other kinds of errors are returned? Should we push this logic into the server type? Maybe something in this method https://github.com/influxdata/kapacitor/blob/master/server/server.go#L882. Plus a method on the Server type to reload.

Reply:
Not totally sure I follow. I think this logic might belong here. `Load` can return few different error types, but there's a configuration setting that specifies what should happen to the runtime in the case of an error. If the configuration says crash on error we wrap any errors coming from `Load` the error with a `load.HardError`.

Reply:
Fixed